assets/content/cookbook/Advanced/10.ScriptedLevels.md

```diff
@@ -0,0 +1,65 @@
+[tags]: / "advanced,hscript,level"
+
+# Scripted Levels
+
+This chapter will walk you through the process of adding a script to a Story Menu Level, and giving examples of the kind of custom behavior which can be implemented with this functionality.
+
+Start by creating a scripted class file with the `.hxc` extension (in the `mods/mymod/scripts/levels` if you want to keep things organized).
+
+```haxe
+// Remember to import each class you want to reference in your script!
+import funkin.ui.story.Level;
+
+// Choose a name for your scripted class that will be unique, and make sure to specifically extend the Level class.
+// This class's functions will override the default behavior for the level.
+class BonusWeekLevel extends Level {
+    public function new() {
+        // The constructor gets called once, when the game loads.
+        // The constructor takes one parameter, which is the level ID for the level you are applying the script to.
+		super('bonusWeek');
+	}
+
+    // Add override functions here!
+}
+```
+
+You can then add override functions to perform custom behavior.
+
+## Hiding specific Songs from the Menu
+
+By overriding the function `getSongDisplayNames`, you can hide certain songs from showing up on the story menu until a certain criteria is met. An example for this can be found in the script file for the Weekend 1 level.
```

Comment:
EMBEDDING: End of line, you should include the footnote you defined at the bottom of the file.